curta/src/chip/ec/edwards/sqrt.rs

```diff
@@ -0,0 +1,236 @@
+use core::str::FromStr;
+
+use num::BigUint;
+use serde::{Deserialize, Serialize};
+
+use super::ed25519::Ed25519BaseField;
+use crate::air::AirConstraint;
+use crate::chip::builder::AirBuilder;
+use crate::chip::field::mul::FpMulInstruction;
+use crate::chip::field::parameters::FieldParameters;
+use crate::chip::field::register::FieldRegister;
+use crate::chip::instruction::Instruction;
+use crate::chip::register::array::ArrayRegister;
+use crate::chip::register::u16::U16Register;
+use crate::chip::register::RegisterSerializable;
+use crate::chip::trace::writer::TraceWriter;
+use crate::chip::utils::digits_to_biguint;
+use crate::chip::AirParameters;
+use crate::math::prelude::*;
+use crate::polynomial::parser::PolynomialParser;
+use crate::polynomial::to_u16_le_limbs_polynomial;
+
+/// Fp Square Root. Computes `sqrt(a) = result`.
+///
+/// This is done by witnessing the square root and then constraining that result * result == a.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct FpSqrtInstruction {
```

Comment:
Maybe we should call it Ed25519Sqrt to indicate it's only for this field and leave the name free for a general square root instruction

Reply:
ya, makes sense.